subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/PropertyReportTask.java

```diff
@@ -42,7 +61,17 @@ public void setRenderer(PropertyReportRenderer renderer) {
 
     @Override
     public void generate(Project project) {
-        for (Map.Entry<String, ?> entry : new TreeMap<String, Object>(project.getProperties()).entrySet()) {
+        Map<String, Object> entries = new TreeMap<>(project.getProperties());
```

Comment:
Is there any reason why this is:
 - A `TreeMap`, instead of a `HashMap`? Sorting?
 - And why we are copying the entire map, meanwhile we do not pass it.